### PR TITLE
GH-227 adds all tags available from kafka metrics

### LINKED ISSUE #227: Feature Request: Add More Metrics Tags
Add `topic` tag to list of included tags for kafka metrics.

For consumers, also add the `group-id` tag.

https://kafka.apache.org/21/javadoc/?org/apache/kafka/common/MetricName.html

### PR DESCRIPTION
**Pull request by timyates**
Merges master in to https://github.com/micronaut-projects/micronaut-kafka/pull/410

Checked with JAPICMP and all looks good (the change is binary compatible)

image

Fixes #227

### CODE REVIEW

File: kafka/src/main/java/io/micronaut/configuration/kafka/metrics/ConsumerKafkaMetricsReporter.java [hunk 1]
```diff
@@ -38,6 +40,13 @@ protected String getMetricPrefix() {
         return CONSUMER_PREFIX;
     }
 
+    @Override
+    protected Set<String> getIncludedTags() {
+        HashSet<String> tags = new HashSet<>(super.getIncludedTags());
+        tags.add("partition");
```
Comment (sdelamo) [inline kafka/src/main/java/io/micronaut/configuration/kafka/metrics/ConsumerKafkaMetricsReporter.java:46]: extract-constant

File: kafka/src/main/java/io/micronaut/configuration/kafka/metrics/AbstractKafkaMetricsReporter.java [hunk 2]
```diff
@@ -115,7 +115,11 @@ private void registerMetric(MeterRegistry meterRegistry, KafkaMetric metric) {
      * @return The tags to include
      */
     protected Set<String> getIncludedTags() {
-        return Collections.singleton("client-id");
+        HashSet<String> tags = new HashSet<>();
+        tags.add("client-id");
```
Comment (sdelamo) [inline kafka/src/main/java/io/micronaut/configuration/kafka/metrics/AbstractKafkaMetricsReporter.java:119]: extract constants

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Comment by timyates** [comment]
@sdelamo done, and updated japicmp image above

**Comment by timyates** [comment]
@dstepanov Hiya! 👋 I noticed you recently made changes in the tests that are failing...  Do you have an idea about how to fix the flakiness?

**Comment by timyates** [comment]
Asked a question as to why we are not seeing a metric used in the tests

https://github.com/micronaut-projects/micronaut-kafka/pull/410#pullrequestreview-918549554

File: kafka/src/test/groovy/io/micronaut/configuration/kafka/AbstractEmbeddedServerSpec.groovy [hunk 3]
```diff
@@ -30,11 +36,6 @@ abstract class AbstractEmbeddedServerSpec extends AbstractKafkaSpec {
         kafkaContainer.start()
     }
 
-    void cleanupSpec() {
-        context.stop()
-        kafkaContainer.stop()
-    }
-
```
Comment (timyates) [inline kafka/src/test/groovy/io/micronaut/configuration/kafka/AbstractEmbeddedServerSpec.groovy:37]: This shouldn't be required as they're already marked with `@AutoCleanup`

**Comment by timyates** [comment]
@sdelamo All green 😅  ready for review

**Pull request merged**